### PR TITLE
[114] Add CI for A11y (axe-core-rspec)

### LINKED ISSUE #114: Continuous a11y integration: axe
As a site engineer, in order to ensure the quality and accessibility of our code, I would like an automated accessibility tool to test all our of code.

Acceptance criteria:

- [ ] [axe-core-rspec](https://github.com/dequelabs/axe-core-gems/tree/develop/packages/axe-core-rspec) is implemented for Challenge_platform repo and runs with the other CI tests

### PR DESCRIPTION
**Pull request by stepchud**
This PR addresses https://github.com/GSA/Challenge_platform/issues/114

### CODE REVIEW

File: app/views/dashboard/index.html.erb [hunk 1]
```diff
@@ -1,9 +1,11 @@
-<div class="usa-card__container custom-card col-md-6 rules-of-behavior">
-  <div class="usa-card__body">
-    <% if logged_in? %>
-      Logged In Dashboard Index
-    <% else %>
-      Logged Out Dashboard Index
-    <% end %>
+<main role="main">
+  <div class="usa-card__container custom-card col-md-6 rules-of-behavior">
+    <h1 class="usa-card__body">
```
Comment (stepchud) [inline app/views/dashboard/index.html.erb:3]: These new lines fix some existing a11y issues

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <!DOCTYPE html>
-<html>
+<html lang="en">
```
Comment (stepchud) [inline app/views/layouts/application.html.erb:2]: a11y fix

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -34,5 +32,7 @@
 
     <%= render "layouts/footer" %>
     <%= render "modals/renew_session" %>
+    <%= javascript_include_tag 'uswds', async: true %>
+    <%= javascript_include_tag 'uswds-init', async: true %>
```
Comment (stepchud) [inline app/views/layouts/application.html.erb:36]: fix the load order bug that caused race condition on js eval

File: app/views/modals/_renew_session.html.erb [hunk 4]
```diff
@@ -1,18 +1,15 @@
-<a href="#renew-modal" style="display:none;" id="renew-modal-open-button" class="usa-button" aria-controls="renew-modal" data-open-modal></a>
-
-<div
-  class="usa-modal"
-  id="renew-modal"
-  aria-labelledby="modal-1-heading"
-  aria-describedby="modal-1-description"
->
+<dialog
```
Comment (stepchud) [inline app/views/modals/_renew_session.html.erb:1]: switching to standard `<dialog>` component allows us to bypass the hidden button clicks that were blocking keyboard tab navigation a11y

**Review by cpreisinger — APPROVED** [review]

**Pull request merged**